### PR TITLE
feat(editor): Add missing extension methods for expressions

### PR DESCRIPTION
**Pull request by elsmr**
## Summary

- String
  - 🆕 toDateTime()
  - ❌ toDate()
  - 🆕 parseJson()
  - 🆕 extractUrlPath()
  - 🆕 toBoolean()
  - 🆕 base64Encode()
  - 🆕 base64Decode()
- Number
  - 🆕 toDateTime()
  - 🆕 toBoolean()
  - 🆕 toString() (native)
  - 🆕 toInt/toFloat (*)
- Object
  - 🆕 toJsonString()
  - 🆕 toDateTime/toString/toBoolean/toInt/toFloat (**)
- Array
  - 🆕 toJsonString()
  - 🆕 toDateTime/toString/toBoolean/toInt/toFloat (**)
- Date & DateTime
  - 🆕 toDateTime()
  - 🆕 toInt()
  - 🆕 toFloat() (*)
  - 🆕 toBoolean() (**)
- Boolean
  - 🆕 toString() (native)
  - 🆕 toInt()
  - 🆕 toFloat/toBoolean (*)
  - 🆕 toDateTime (**)
  
> 🆕 new extension method
> ❌ hidden from autocomplete, still works for backwards compatibility
> (\*) hidden from autocomplete, but still defined to work with mixed data ⬇️ 
> (\*\*) always returns undefined, but should not throw with mixed data ⬇️ 

```typescript
const items = [
  { json: { mixed: 'true' } },
  { json: { mixed: 1 } },
  { json: { mixed: true } },
  { json: { mixed: {} }
}]

{{ $json.mixed.toBoolean() }} // no error
```

Specs in Notion

## Related tickets and issues
https://www.notion.so/n8n/Autocomplete-review-4ab890a4da8243b186c4a3560a3997d5#d35790a5cc1e4d48902b945bbceabd01
https://linear.app/n8n/issue/NODE-1183/add-missing-extension-methods

## Review / Merge checklist
- [ ] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included.
   > A bug is not considered fixed, unless a test is added to prevent it from happening again.
   > A feature is not complete without tests.

### CODE REVIEW

**Comment by pemontto** [comment]
:pray: b64 encode and decode for strings :pray:

https://github.com/n8n-io/n8n/pull/7762

**Comment by elsmr** [comment]
> 🙏 b64 encode and decode for strings 🙏
> 
> #7762

Would love to include those 😄 I should have time next week to finish this PR

File: packages/editor-ui/src/plugins/codemirror/completions/datatype.completions.ts
Comment (MiloradFilipovic) [inline packages/editor-ui/src/plugins/codemirror/completions/datatype.completions.ts:564]: Is this supposed to be case sensitive?

File: packages/editor-ui/src/plugins/codemirror/completions/datatype.completions.ts
Comment (MiloradFilipovic) [inline packages/editor-ui/src/plugins/codemirror/completions/datatype.completions.ts:594]: Is this a requirement to only allow 30 years in the future? Can we use whole [possible](https://262.ecma-international.org/5.1/#sec-15.9.1.1) range for date here?

**Review by MiloradFilipovic — DISMISSED** [review]
Looks really good. Left a few minor comments but good to go otherwise

File: packages/editor-ui/src/plugins/codemirror/completions/datatype.completions.ts
Comment (elsmr) [reply packages/editor-ui/src/plugins/codemirror/completions/datatype.completions.ts:594]: The problem would become that we would suggest `toDateTime` on any number. The idea here is that we only suggest it when we are pretty sure it's a timestamp (so within a range of now) but the 30y is arbitrary.

**Review by MiloradFilipovic — DISMISSED** [review]
Looks good, thanks for addressing!

**Review by MiloradFilipovic — APPROVED** [review]
Re-approving after merging in `master`

**Comment by janober** [comment]
Got released with [`n8n@1.34.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.34.0)

**Pull request merged**
